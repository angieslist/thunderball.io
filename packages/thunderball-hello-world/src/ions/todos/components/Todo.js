import React from 'react';
import PropTypes from 'prop-types';

const Todo = ({ onClick, id, completed, text }) => (
  <div>
    <input
      type="checkbox"
      id={`todo-${id}`}
      onChange={onClick}
      checked={completed}
    />
    <label
      className="label-inline"
      htmlFor={`todo-${id}`}
    >
      {text}
    </label>
  </div>
);

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};

export default Todo;
