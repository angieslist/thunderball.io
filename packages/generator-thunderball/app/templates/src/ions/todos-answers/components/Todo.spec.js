import React from 'react';
import { shallow } from 'enzyme';

import Todo from './Todo';

const setup = (setupProps = {}) => {
  const defaultProps = {
    onClick: jest.fn(),
    id: 123,
    completed: false,
    text: 'Test todo',
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <Todo
      {...props}
    />,
  );

  return {
    props,
    wrapper,
  };
};

describe('Todo', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('marks as checked when completed', () => {
    const { wrapper } = setup({ completed: true });

    expect(wrapper).toMatchSnapshot();
  });

  test('calls onClick when changed', () => {
    const { props, wrapper } = setup();

    wrapper.find('input').simulate('change');
    expect(props.onClick).toHaveBeenCalled();
  });
});
