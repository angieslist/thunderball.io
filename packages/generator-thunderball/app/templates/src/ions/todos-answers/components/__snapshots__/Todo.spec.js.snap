// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo marks as checked when completed 1`] = `
<div>
  <input
    checked={true}
    id="todo-123"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <label
    className="label-inline"
    htmlFor="todo-123"
  >
    Test todo
  </label>
</div>
`;

exports[`Todo renders without crashing 1`] = `
<div>
  <input
    checked={false}
    id="todo-123"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <label
    className="label-inline"
    htmlFor="todo-123"
  >
    Test todo
  </label>
</div>
`;
